### PR TITLE
Pd/fix 6

### LINKED ISSUE #6: Value of attribute cannot be set to `nil` via `#update`
When calling `update` on a Valued object and passing `nil` as a new value, it is not set to `nil` but instead remains at the current value

```ruby
class Quantity
  include Valued

  attributes :unit, :amount
end
```
```ruby
quantity = Quantity.new(unit: 'kg', amount: 1)
=> #<Quantity unit="kg" amount=1>
quantity.update(amount: nil)
=> #<Quantity unit="kg" amount=1>
```

I would expect that I can set the value to `nil`.

### PR DESCRIPTION
**Pull request by pderichs**
Hi,

this PR tries to fix https://github.com/mmainz/valued/issues/6.

I have also extracted shared samples within the specs (let me know, if you would like to have a separate PR for that). 

`BUNDLED WITH` was updated as well in the process. I am unsure about this change. I have left it in this PR to get your opinion on that.

Let me know what you think :slightly_smiling_face: 

Cheers,
Patrick

### CODE REVIEW

File: valued.gemspec [hunk 1]
```diff
@@ -17,10 +17,12 @@ Gem::Specification.new do |spec|
...
     Dir.chdir(File.expand_path('..', __FILE__)) do
-      `git ls-files -z`.split("\x0").reject do |f|
-        f.match(%r{^(test|spec|features)/})
-      end
+    `git ls-files -z`.split("\x0").reject do |f|
+      f.match(%r{^(test|spec|features)/})
     end
+  end
+  spec.files += `git ls-files -- spec/valued/shared_examples/*.rb`.split("\n")
+
```
Comment (mmainz) [inline valued.gemspec]: Shouldn't this be included in the above? Why do we need to add this explicitly?

File: spec/valued/mutable_spec.rb [hunk 2]
```diff
@@ -1,149 +1,22 @@
+require 'spec_helper'
```
Comment (mmainz) [inline spec/valued/mutable_spec.rb]: You don't need to require `spec_helper`, it is automatically required by RSpec.

**Review by mmainz — COMMENTED** [review]
Looks good overall! Just some minor things. Regarding the `BUNDLED_WITH`, I think updating that is fine. If some one doesn't have that bundler version, that person should just update.

But I think you messed up my preferred formatting. I usually format my code with `@prettier/plugin-ruby`. But I also saw that I don't have a CI step for this. Also, I'm still using Travis instead of the vastly superior GitHub Actions. I'll add that to master so that you can rebase and get the formatting right.

Thanks a lot for fixing this!

File: valued.gemspec (hunk 1 above)
Comment (pderichs) [reply valued.gemspec]: You are right :facepalm: Removed.

File: spec/valued/mutable_spec.rb (hunk 2 above)
Comment (pderichs) [reply spec/valued/mutable_spec.rb]: `true`. Removed :+1:

**Comment by mmainz** [comment]
@pderichs after some trial-and-error, I succeeded in adding a GitHub Action that tests and checks the code format. To format you code, just do `npm install` and then `npm run format`.

**Comment by pderichs** [comment]
@mmainz Thanks for all your effort and hints! I have updated the PR accordingly.

**Pull request merged**